spec/loaders/doppler_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Anyway::Loaders::Doppler do
+  subject { described_class.call(**options) }
+
+  let(:options) { {env_prefix: "vne", some_other: "value"} }
+
+  context "when `DOPPLER_TOKEN` env variable is not specified" do
+    before do
+      ENV["DOPPLER_TOKEN"] = nil
+    end
+
+    it "raises an error" do
+      expect { described_class.call(**options) }.to raise_error("Please specify `DOPPLER_TOKEN` env variable")
+    end
+  end
+
+  context "when the API request fails" do
+    before do
+      ENV["DOPPLER_TOKEN"] = "token"
+      response = double(Net::HTTPResponse)
+      allow(response).to receive(:code).and_return("404")
+      allow(response).to receive(:body).and_return({"messages" => ["some error"]}.to_json)
+      allow(Net::HTTP).to receive(:get_response).and_return(response)
```

Comment:
It's better to avoid such testing style, 'cause it literally duplicates the source code, not the behaviour. Tools like [webmock](https://github.com/bblimke/webmock) is a way to go.